### PR TITLE
Don't get stuck in programmatic scrolling

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt [hunk 1]
```diff
@@ -169,7 +177,6 @@ internal open class UIViewLazyList(
     tableView.apply {
       dataSource = this@UIViewLazyList.dataSource
       delegate = tableViewDelegate
-      prefetchingEnabled = true
```
Comment (swankjesse) [inline redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt:172]: This doesn’t work on iOS 14

File: samples/emoji-search/ios-uikit/EmojiSearchApp/Info.plist [hunk 2]
```diff
@@ -4,6 +4,8 @@
 <dict>
 	<key>NSAppTransportSecurity</key>
 	<dict>
+		<key>NSAllowsArbitraryLoads</key>
+		<true/>
```
Comment (swankjesse) [inline samples/emoji-search/ios-uikit/EmojiSearchApp/Info.plist:8]: This is required for iOS 14

**Review by swankjesse — COMMENTED** [review]
I saw that our scroll position wasn’t tracking properly.

It turns out that we’d programmatically scrolled to position 0. When you do that and you’re already at position 0, then `scrollViewDidEndScrollingAnimation()` is never called.

File: redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt (hunk 1 above)
Comment (swankjesse) [reply redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt:172]: Also defaults to YES anyway!
https://developer.apple.com/documentation/uikit/uitableview/3801922-prefetchingenabled

File: redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt [hunk 3]
```diff
@@ -160,6 +160,14 @@ internal open class UIViewLazyList(
         scrollProcessor.onUserScroll(firstIndex, lastIndex)
       }
 
+      /**
+       * If the user begins a drag while we’re programmatically scrolling, well then we're not
+       * programmatically scrolling anymore.
+       */
+      override fun scrollViewWillBeginDragging(scrollView: UIScrollView) {
+        isDoingProgrammaticScroll = false
```
Comment (dnagler) [inline redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt:168]: Yep, this sounds right to me

**Review by dnagler — APPROVED** [review]

**Review by JakeWharton — APPROVED** [review]

**Pull request merged**